js/packages/openinference-instrumentation-openai/src/instrumentation.ts

```diff
@@ -409,6 +410,24 @@ function getLLMInputMessagesAttributes(
   }, {} as Attributes);
 }
 
+/**
+ * Converts each tool definition into a json schema
+ */
+function getLLMToolsJSONSchema(
+  body: ChatCompletionCreateParamsBase,
+): Attributes {
+  if (!body.tools) {
+    // If tools is undefined, return an empty object
+    return {} as Attributes;
+  }
+  return body.tools.reduce((acc, tool, index) => {
+    const toolJsonSchema = JSON.stringify(tool);
+    const key = `${SemanticConventions.LLM_TOOLS}.${index}.${SemanticConventions.TOOL_JSON_SCHEMA}`;
+    acc[key] = toolJsonSchema;
+    return acc;
+  }, {} as Attributes);
```

Comment:
i think you can put this type on line 423 instead of casting it down here, acc: Attributes - sometimes that seems to work with reduce sometimes it can be finicky, if you can would do that to avoid the cast here

Reply:
Done